### PR TITLE
Add options to the properties converter to update from 0.6.x to 0.7.x

### PR DESCRIPTION
**Pull request by drcrallen**
* Add the -c option to specify one of FIVE_TO_SIX or SIX_TO_SEVEN

### CODE REVIEW

**Comment by drcrallen** [comment]
CI failed only on java 8 on
```
Failed tests:   testQueryCancellation(io.druid.query.ChainedExecutionQueryRunnerTest)
```

File: services/src/main/java/io/druid/cli/convert/ConvertProperties.java [hunk 1]
```diff
@@ -125,12 +136,111 @@
...
+
   @Option(name = "-f", title = "file", description = "The properties file to convert", required = true)
   public String filename;
 
   @Option(name = "-o", title = "outFile", description = "The file to write updated properties to.", required = true)
   public String outFilename;
 
+  @Option(name = "-c", title = "conversionGroup", description = "The conversion group to use for converting properties", required = true, allowedValues = {
+      "FIVE_TO_SIX",
+      "SIX_TO_SEVEN"
```
Comment (xvrl) [inline services/src/main/java/io/druid/cli/convert/ConvertProperties.java]: I think we should default to SIX_TO_EVEN or even remove the 0.5->0.6 code altogether

Comment (drcrallen) [inline services/src/main/java/io/druid/cli/convert/ConvertProperties.java]: Removed

**Comment by xvrl** [comment]
We also need to add conversion for cache settings

**Comment by xvrl** [comment]
```properties
druid.announcer.type=batch
druid.curator.compress=true
```
settings can be removed if set to those specific values, since they are now the defaults.

```properties
druid.db.connector.useValidationQuery
```
should be removed, since it does not exist anymore and is always true

**Comment by drcrallen** [comment]
Victim of #1032

File: services/src/main/java/io/druid/cli/convert/ConvertProperties.java [hunk 2]
```diff
@@ -35,94 +42,103 @@
...
-      new Rename("druid.pusher.cassandra.host", "druid.storage.host"),
-      new Rename("druid.pusher.cassandra.keySpace", "druid.storage.keySpace")
+  private static String parseJdbcUrl(String jdbcUrl){
+    final String subString = jdbcUrl.substring("jdbc:".length());
+    final URI uri = URI.create(subString);
+    return uri.getScheme();
+  }
+
+  private static final List<PropertyConverter> converters6to7 = ImmutableList.<PropertyConverter>of(
+      new Rename("druid.db.connector.ConnectURI", "druid.metadata.storage.connector.connectURI"),
```
Comment (fjy) [inline services/src/main/java/io/druid/cli/convert/ConvertProperties.java]: be careful of casing here

Comment (drcrallen) [inline services/src/main/java/io/druid/cli/convert/ConvertProperties.java]: fixed

**Pull request merged**